### ISSUE TITLE
Document Explicit Branch Workflow Dispatch Reruns

### ISSUE THREAD

**Issue by klondikemarlen** [klondikemarlen/marlens-skills-rules-and-tools#370]
## Problem

After pushing a fix to a pull-request branch, agents can inspect a stale automatic run or assume a push retriggered all checks. That leaves branch-specific verification ambiguous.

## Evidence

During [WRAP PR #584](https://github.com/icefoganalytics/wrap/pull/584), the user explicitly corrected the workflow twice: rerun the Test Suite for the branch directly. The reliable command was:

```bash
gh workflow run test.yml \
  --repo icefoganalytics/wrap \
  --ref improve-workflow-search-performance
```

The resulting run must then be watched by its run ID. This avoided relying on an older failed run attached to a previous branch head.

## Proposed Guidance

For repositories whose test workflow declares `workflow_dispatch`:

1. Trigger the workflow explicitly with `gh workflow run <workflow> --ref <branch>` after a corrective push.
2. Identify the newly queued run and verify its `headSha` matches the intended branch head.
3. Watch that exact run to completion; do not report a prior run on the branch as current evidence.

## Acceptance Criteria

- Pull-request or CI guidance documents explicit branch-ref dispatch as the rerun path when `workflow_dispatch` is available.
- Guidance requires checking the resulting run's branch and head SHA before reporting success.
- Guidance retains the existing distinction between a completed CI result and a currently running or stale run.

## Duplicate Check

Closed issue #231 covers adding CI, not branch-specific manual rerun verification.
